src/EventStore.Client/SingleNodeChannelSelector.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+
+namespace EventStore.Client {
+	internal class SingleNodeChannelSelector : IChannelSelector {
+		private readonly ChannelBase _channel;
+		private readonly Lazy<Task<ServerCapabilities>> _serverCapabilitiesLazy;
+		private Task<ServerCapabilities> ServerCapabilities => _serverCapabilitiesLazy.Value;
+
+		public SingleNodeChannelSelector(EventStoreClientSettings settings,
+			IServerCapabilitiesClient serverCapabilities,
+			CancellationToken cancellationToken) {
+			_channel = ChannelFactory.CreateChannel(settings, settings.ConnectivitySettings.Address);
+			_serverCapabilitiesLazy =
```

Comment:
consider

```
			_serverCapabilitiesLazy = new Lazy<Task<ServerCapabilities>>(() =>
				serverCapabilities.GetAsync(_channel, cancellationToken).AsTask());
```
could be better not to pass LazyThreadSafetyMode in the other Lazy uses too for simplicity